code/control/DeployDispatcher.php

```diff
@@ -76,31 +77,54 @@ public function history(SS_HTTPRequest $request) {
 		$start = ($page - 1) * $list->getPageLength();
 		$list->setPageStart((int) $start);
 		if (empty($list)) {
-			return $this->getAPIResponse(['message' => 'No deploy history'], 404);
+			return $this->getAPIResponse(['list' => []], 200);
 		}
 
+		$currentBuild = $this->environment->CurrentBuild();
+
 		foreach ($list as $deployment) {
 			$data[] = [
+				'ID' => $deployment->ID,
 				'CreatedDate' => $deployment->Created,
```

Comment:
I think upper camel case for now, so that it matches the current DNDeployment etc field names in the job models.